c/tests/pn_test.hpp

```diff
@@ -46,13 +46,14 @@ namespace pn_test {
 template <class T, void (*Free)(T *)> class auto_free {
   T *ptr_;
   auto_free &operator=(auto_free &x);
-  auto_free(auto_free &x);
+public:
+  auto_free(auto_free &x) = delete;
 
 public:
-  auto_free(T *p = 0) : ptr_(p) {}
+  explicit auto_free(T *p = 0) : ptr_(p) {}
   ~auto_free() { Free(ptr_); }
   T *get() const { return ptr_; }
-  operator T *() const { return ptr_; }
+  operator T *() const { return ptr_; } // not marking explicit for convenience
```

Comment:
This class is something like an `std::unique_ptr` with a deleter. Apparently having explicit `operator*` was considered and dismissed, for reasons https://herbsutter.com/2012/06/21/reader-qa-why-dont-modern-smart-pointers-implicitly-convert-to/

Reply:
Actually it could be replaced with unique_ptr now - this version will work with C++03 which isn't needed any more. [std::unique_ptr has an optional deleter capability]

Reply:
Nut not having explicit here is the opposite of the advice from Herb Sutter I think.

Reply:
Ok, I edited my comment too late, now the conversation does not make sense. Revert!

Reply:
Honestly I'm fine with putting this PR in as is and then moving it closer to good C++11 piece-by-piece